### PR TITLE
[MI-3466] Fix Jira issue #855: Save user's last used field values

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
Link: https://github.com/mattermost/mattermost-plugin-jira/issues/855

### CODE REVIEW

File: webapp/src/components/jira_instance_and_project_selector/jira_instance_and_project_selector.test.tsx [hunk 1]
```diff
@@ -26,7 +26,9 @@ describe('components/JiraInstanceAndProjectSelector', () => {
         connectedInstances: [{instance_id: 'instance1', type: InstanceType.CLOUD}, {instance_id: 'instance2', type: InstanceType.SERVER}],
         defaultUserInstanceID: '',
         fetchJiraProjectMetadata: jest.fn().mockResolvedValue({data: {
-            default_project_key: 'TEST',
+            default_field_values: {
+                project_key: 'TEST',
```
Comment (ayusht2810) [inline webapp/src/components/jira_instance_and_project_selector/jira_instance_and_project_selector.test.tsx:30]: should we include other fields as well?

File: webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.tsx [hunk 2]
```diff
@@ -244,10 +244,11 @@ export default class EditChannelSubscription extends PureComponent<Props, State>
         }
 
         this.setState({instanceID, error: null});
-        this.handleProjectChange('');
+        this.handleProjectChange({});
     }
 
-    handleProjectChange = (projectID: string) => {
+    handleProjectChange = (fieldValues: DefaultFieldValues) => {
+        const projectID = fieldValues.project_key ? fieldValues.project_key : '';
```
Comment (ayusht2810) [inline webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.tsx:251]:
```suggestion
        const projectID = fieldValues.project_key ?? '';
```

Comment (ayusht2810) [reply webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.tsx:251]: check if this works. fix everywhere if updating

Comment (raghavaggarwal2308) [reply webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.tsx:251]: We do not have the setup for nullish coleasing in Jira

File: webapp/src/components/jira_instance_and_project_selector/jira_instance_and_project_selector.test.tsx (hunk 1 above)
Comment (raghavaggarwal2308) [reply webapp/src/components/jira_instance_and_project_selector/jira_instance_and_project_selector.test.tsx:30]: No, only these fields were requested for now. Will update if asked.

**Review by ayusht2810 — APPROVED** [review]

File: server/command.go [hunk 3]
```diff
@@ -1035,8 +1035,8 @@ func executeMe(p *Plugin, c *plugin.Context, header *model.CommandArgs, args ...
 
 			resp += connectionBullet(info.User.ConnectedInstances.Get(instanceID), connection, info.User.DefaultInstanceID == instanceID)
 			resp += fmt.Sprintf("   * %s\n", connection.Settings)
-			if connection.DefaultProjectKey != "" {
-				resp += fmt.Sprintf("   * Default project: `%s`\n", connection.DefaultProjectKey)
+			if connection.DefaultFieldValues != nil && connection.DefaultFieldValues.ProjectKey != "" {
```
Comment (Kshitij-Katiyar) [inline server/command.go:1038]: change this variable name

File: webapp/src/components/modals/attach_comment_modal/attach_comment_form.tsx [hunk 4]
```diff
@@ -94,7 +94,7 @@ export default class AttachCommentToIssueForm extends PureComponent<Props, State
                 selectedProjectID={''}
                 hideProjectSelector={true}
                 onInstanceChange={(instanceID: string) => this.setState({instanceID})}
-                onProjectChange={(projectKey: string) => {}}
+                onProjectChange={(_: DefaultFieldValues) => {}}
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/attach_comment_modal/attach_comment_form.tsx:97]: check this

File: webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.test.tsx [hunk 5]
```diff
@@ -138,7 +138,9 @@ describe('components/EditChannelSubscription', () => {
             <EditChannelSubscription {...props}/>
         );
         wrapper.setState(baseState);
-        wrapper.instance().handleProjectChange('TES');
+        wrapper.instance().handleProjectChange({
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.test.tsx:141]: try to add test cases here for the other values of this object.

File: webapp/src/components/modals/create_issue/create_issue_form.tsx [hunk 6]
```diff
@@ -132,21 +133,32 @@ export default class CreateIssueForm extends React.PureComponent<Props, State> {
...
-        fields.issuetype = {
-            id: issueType,
-        };
+        if (fieldValues.issue_type) {
+            fields.issuetype = {
+                id: fieldValues.issue_type,
+            };
+        } else {
+            const issueTypes = getIssueTypes(this.state.jiraIssueMetadata, projectKey);
+            const issueType = issueTypes.length ? issueTypes[0].id : '';
```
Comment (Kshitij-Katiyar) [inline webapp/src/components/modals/create_issue/create_issue_form.tsx:148]: Please check whether we need 0 index every time

Comment (raghavaggarwal2308) [reply webapp/src/components/modals/create_issue/create_issue_form.tsx:148]: yes

File: webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.test.tsx (hunk 5 above)
Comment (raghavaggarwal2308) [reply webapp/src/components/modals/channel_subscriptions/edit_channel_subscription.test.tsx:141]: Not required

**Review by Kshitij-Katiyar — APPROVED** [review]

**Pull request merged**